src/main/java/com/uber/profiling/YamlConfigProvider.java

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright (c) 2018 Uber Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.uber.profiling;
+
+import com.uber.profiling.util.AgentLogger;
+import com.uber.profiling.util.IOUtils;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.yaml.snakeyaml.Yaml;
+
+import java.io.ByteArrayInputStream;
+import java.net.HttpURLConnection;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class YamlConfigProvider implements ConfigProvider {
+    private static final AgentLogger logger = AgentLogger.getLogger(YamlConfigProvider.class.getName());
+    
+    private static final String OVERRIDE_KEY = "override";
+    
+    private String filePath;
+
+    public YamlConfigProvider() {
+    }
+    
+    public YamlConfigProvider(String filePath) {
+        this.filePath = filePath;
+    }
+
+    public void setFilePath(String filePath) {
+        this.filePath = filePath;
+    }
+
+    @Override
+    public Map<String, Map<String, List<String>>> getConfig() {
+        return getConfig(filePath);
+    }
+    
+    private Map<String, Map<String, List<String>>> getConfig(String filePath) {
+        if (filePath == null || filePath.isEmpty()) {
+            logger.warn("Empty YAML config file path");
+            return new HashMap<>();
+        }
+
+        byte[] bytes;
+        
+        try {
+            String filePathLowerCase = filePath.toLowerCase();
+            if (filePathLowerCase.startsWith("http://") || filePathLowerCase.startsWith("https://")) {
```

Comment:
try to enforce `https:` if we can..

Reply:
It may use http in some situations. For example, if a company uses it in their internal datacenter, and may have the config file placed by http.